service/controller/resource/certificates/create.go

```diff
@@ -0,0 +1,39 @@
+package certificates
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	targetSecret, err := toTargetSecret(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	sourceSecret, err := toSourceSecret(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	_, err = r.k8sClient.K8sClient().CoreV1().Secrets(targetSecret.GetNamespace()).Get(targetSecret.GetName(), metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
```

Comment:
What if source secret is rolled?

Reply:
Do you mean updated? That's no handled.

Reply:
Isn't that a problem?

Reply:
so far we are good with this. We can improve it later.